msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/DefaultHttpClient.java

```diff
@@ -117,6 +118,10 @@ private HttpResponse readResponseFromConnection(final HttpsURLConnection conn) t
             httpResponse.addHeaders(conn.getHeaderFields());
             httpResponse.body(inputStreamToString(is));
             return httpResponse;
+        } catch (SocketTimeoutException readException) {conn.getURL();
```

Comment:
Changing exception behavior is tricky because exceptions are part of the public api. I would think many apps have retry policies that look for `SocketTimeout` and `Connect` exceptions, and this change will break the retry policies.

Maybe log.error these messages, but don't change exception type.

Reply:
Good point, in the latest commit I log the message explaining the error but the original exception gets re-thrown.